src/setup/destroy.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  DeleteBucketCommand,
+  DeleteObjectCommand,
+  ListObjectsCommand,
+} from "@stedi/sdk-client-buckets";
+import { bucketClient } from "../lib/buckets.js";
+import { guidesClient } from "../support/guide.js";
+import { stashClient } from "../lib/stash.js";
+import { DeleteGuideCommand } from "@stedi/sdk-client-guides";
+import {
+  PARTNERS_KEYSPACE_NAME,
+  CONTROL_NUMBER_KEYSPACE_NAME,
+} from "../lib/constants.js";
+import { DeleteKeyspaceCommand } from "@stedi/sdk-client-stash";
+import * as dotenv from "dotenv";
+import { DeleteUserCommand } from "@stedi/sdk-client-sftp";
+import { sftpClient } from "../lib/sftp.js";
+import { functionClient } from "../support/functions.js";
+import { DeleteFunctionCommand } from "@stedi/sdk-client-functions";
+import { updateDotEnvFile } from "../support/utils.js";
+dotenv.config();
+
+(async () => {
+  console.log("Deleting all resources provisioned by bootstrap");
+
+  // Delete Buckets
+  console.log("Deleting Buckets");
+  await emptyAndDeleteBucket(process.env.SFTP_BUCKET_NAME ?? "");
+  await emptyAndDeleteBucket(process.env.EXECUTIONS_BUCKET_NAME ?? "");
+
+  // Delete Guides
+  console.log("Deleting Guides");
+  const guideIds = process.env.GUIDE_IDS?.split(",");
```

Comment:
after thinking about this a bit, I think it might be better to not depend on the `.env` file for cleanup. there are a few reasons why:

- a user might deploy the bootstrap to multiple accounts (we do this all the time for various customers, and customers themselves may have staging and production accounts, for example)
- a different user may want to clean up resources in their account (or even the same user from a different machine)

I know that we already add a stash key/value for migration details, maybe we can expand that to a broader `deploymentMetadata` key that includes the migration details, as well as a list of deployed resources (presumably a map of resource type to id, so that we know how to clean them all up). Certainly open to other ideas here, just don't think the `.env` approach is the best solution.

Reply:
That makes sense, I didn't know about the stash keyspace we already use, how does that get provisioned? Seems like we need somewhere to do state management 🤔

Reply:
looks like the migration status bit is stored here: https://github.com/Stedi-Demos/bootstrap/blob/main/src/lib/migration/stashStorage.ts#L24-L35

by default it uses the partner config keyspace (and I think it would be fine to store bootstrap resource state here as well). the default limit for stash is 10 keyspaces, so probably better to avoid using some other keyspace for tracking state